### DISCUSSION TITLE
How can a failed job be manually retried? retry() usage question

### DISCUSSION THREAD

**Discussion by Ben-CA** [OptimalBits/bull#2466] [Q&A] [answered]
How can the "Job Retry" function be used? - such as if a failed job's id is passed to Bull, then retry the failed job.
https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#jobretry

Is there a code example anywhere of how to use retry()?

And from what I understand, TaskForce can manually retry failed jobs.

**Comment by manast** [comment, marked as answer]
Well, the code example would just be:
```ts
await job.retry();
```
And yes, Taskforce.sh allows manual retrying, and it is using this same method.

**Comment by Ben-CA** [comment]
Thanks @manast ; but how do I pass that a job id to tell it which job to retry?

**Reply by manast** [reply]
You just instantiate the job instance and then retry: https://api.docs.bullmq.io/classes/Job.html#fromId

**Reply by Ben-CA** [reply]
So fromId works in Bull "classic" as well as BullMQ?

It's not in the reference anywhere for Bull...
https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#jobretry

**Reply by manast** [reply]
https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#queuegetjob

**Reply by Ben-CA** [reply]
Thanks!

**Comment by Ben-CA** [comment]
Syntax used:
```
  let job = await Queue.getJob(key)  // Fetch job by ID (Queue here being the name of the Queue)
  await job.retry()  // Retry job
```

**Comment by aditodkar** [comment]
If you want to manually retry specific job then you can integrate [bull board](https://github.com/felixmosh/bull-board) without `readOnly` flag. On dashboard you will get an option for retry if job fails.
